### ISSUE TITLE
[windows beta-1.0.0] Can't use crate with '-' in name for [build-dependencies]

### ISSUE THREAD

**Issue by ceram1** [rust-lang/cargo#1527]
I can't use some crates like 'rust-serialize' in [build-dependencies]
But 'phf_codegen' works.

Cargo.toml
```
[package]
build = "codegen/main.rs"

[dependencies]
rustc-serialize = "*"

[build-dependencies]
phf_codegen = "*"
rustc-serialize = "*"
```

```rust
extern crate rustc_serialize;
```
works in src/main.rs,

but
```rust
extern crate phf_codegen;
extern crate rustc-serialize;
```
in codegen/main.rs says
codegen/main.rs:2:19: 2:20 error: expected one of `;` or `as`, found `-`
codegen/main.rs:2 extern crate rustc-serialize;

and

```rust
extern crate phf_codegen;
extern crate rustc_serialize;
```
says
codegen/main.rs:2:1: 2:30 error: can't find crate for `rustc_serialize`
codegen/main.rs:2 extern crate rustc_serialize;

I'm using cargo from windows 1.0.0-beta installer.
```cmd
cargo --version
cargo 0.0.1-pre-nightly (84d6d2c 2015-03-31) (built 2015-04-01)
```

-----
Moved from 
https://github.com/rust-lang/rust/issues/24463

**Comment by ceram1** [comment]
When I tested with nightly, it seems ok.

```cmd
"C:\Program Files\Rust nightly 1.0\bin\cargo" --version
cargo 0.0.1-pre-nightly (d49b443 2015-04-14) (built 2015-04-15)
```

So I think I need to change issue title.

**Comment by alexcrichton** [comment]
Thanks for the report! It looks like this is fixed in master though, so the bug fix should definitely make it into the 1.0 release. Other than that though I don't think there's much to do on this, so closing.
